storage/scylla_api.c

```diff
@@ -274,58 +276,61 @@ static CassError connect_session(CassSession* session, const CassCluster* cluste
   return rc;
 }
 
-static CassError execute_query(CassSession* session, const char* query) {
+static CassError execute_statement(CassSession* session, CassStatement* statement) {
   CassError rc = CASS_OK;
-  CassStatement* statement = cass_statement_new(query, 0);
-  CassFuture* future = cass_session_execute(session, statement);
-
-  cass_future_wait(future);
+  int cnt = 0;
+  while (cnt < MAX_QUERY_RETRY_NUMBER) {
+    CassFuture* future = cass_session_execute(session, statement);
+    cass_future_wait(future);
+
+    rc = cass_future_error_code(future);
+    if (rc != CASS_OK) {
+      ta_log_warning("Scylla request fail %d times\n", cnt + 1);
+      print_error(future);
+    }
+    cass_future_free(future);
+    if (rc == CASS_OK) {
+      break;
+    }
 
-  rc = cass_future_error_code(future);
-  if (rc != CASS_OK) {
-    print_error(future);
+    cnt++;
+    if (cnt < MAX_QUERY_RETRY_NUMBER) {
+      sleep(RETRY_DELAY_SECONDS);
```

Comment:
I am struggling with what is a better method to ensure re-executing queries with a least 5 seconds interval.
Considering about using `sleep()`, `timerfd()` or other methods.

Reply:
Can you introduce something asynchronous? Signals or thread-specific messaging.